app/controllers/search_controller.rb

```diff
@@ -13,14 +13,24 @@ def index
     @advanced_options_open = false
     @search_results = []
     @json_url = json_link("#{rest_url}/search", {})
+    params[:portals] = params[:portals]&.join(',')
 
     return if @search_query.empty?
 
     params[:pagesize] = "150"
-    results = LinkedData::Client::Models::Class.search(@search_query, params).collection
 
+    set_federated_portals
+
+    params[:ontologies] = nil if federatation_enabled?
+
+    @time = Benchmark.realtime do
+      results = LinkedData::Client::Models::Class.search(@search_query, params)
+      @federation_errors = results[:errors].map{|e| find_portal_name_by_api(e.split(' ').last.gsub('search', ''))}
```

Comment:
create a private function in this controller for these two lines of error handling.

Reply:
Okay

Reply:
done